frontend/packages/knative-plugin/src/utils/create-eventsources-utils.ts

```diff
@@ -90,13 +117,23 @@ export const getKafkaSourceResource = (formData: EventSourceFormData): K8sResour
   return _.merge({}, baseResource, kafkaSource);
 };
 
-export const getEventSourceResource = (formData: EventSourceFormData): K8sResourceKind => {
+const getEventSourceResource = (formData: EventSourceFormData): K8sResourceKind => {
   switch (formData.type) {
-    case EventSources.KafkaSource:
+    case KnownEventSources.KafkaSource:
       return getKafkaSourceResource(formData);
     default:
-      return getEventSourcesDepResource(formData);
+      return createResourceFromForm(formData);
+  }
+};
+
+export const getEventSourcesDepResource = (
+  formData: any,
+  isYamlData?: boolean,
+): K8sResourceKind => {
+  if (isYamlData) {
```

Comment:
Why not make use of `getEventSourceResource` itself to send whatever proper resource data needs to be sent?

Reply:
updated `getEventSourceResource` to handle all event sources